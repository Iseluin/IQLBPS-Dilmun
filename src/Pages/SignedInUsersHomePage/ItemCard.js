import React from 'react'

const ItemCard = ({ imgSrc }) => {
    return (
        <div className="max-w-sm bg-white border border-gray-200 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-700">
            <img className="rounded-t-lg w-full" src={imgSrc} alt="Product-img" />
            <div className="p-5">
                <h5 className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white">Backpack</h5>
                <p className="mb-3 font-normal text-gray-700 dark:text-gray-400">Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.</p>
                <button type='submit' className="inline-flex items-center text-background px-3 py-2 rounded bg-secondary ml-2 hover:bg-primary transition duration-300">
                    Read more
                </button>
            </div>
        </div>

    )
}

export default ItemCard