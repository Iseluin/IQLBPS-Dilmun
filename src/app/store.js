
import {
  configureStore
} from '@reduxjs/toolkit';
import userReducer from '../features/user/userSlice';
import itemReducer from '../features/slices/item'

export const store = configureStore({
  reducer: {
    user: userReducer,
    item: itemReducer
  },
});
