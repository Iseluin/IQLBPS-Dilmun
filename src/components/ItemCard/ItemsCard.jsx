function ItemsCard({ item }) {
  return (
    <div
      data-testid="item-card"
      className="xl:w-80 lg:w-72 md:w-80 w-full bg-background rounded-lg hover:shadow-secondary shadow-gray-400 shadow-lg hover:shadow-2xl duration-300"
    >
      <figure className="relative max-w-sm transition-all cursor-pointer">
        <a href="/">
          <img
            className="rounded-t-lg overflow-hidden object-cover mb-3 lg:h-80 h-80 shadow-md md:w-80 xl:w-full lg:w-72 w-full"
            src={item.file}
            alt="product"
          />
          <figcaption className="absolute top-3 right-1 px-4 text-lg">
            {item.type === 'Donated' ? (
              <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                {item.type}
              </span>
            ) : (
              <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                {item.price}
              </span>
            )}
          </figcaption>
        </a>
      </figure>
      <div className="px-5 pb-5">
        {item.title.length > 15 ? (
          <h5 className="text-3xl pb-2 font-extrabold text-primary">
            {item.title.slice(0, 1).toUpperCase()}
            {item.title.slice(1, 8)} ...
          </h5>
        ) : (
          <h5 className="text-3xl pb-2 font-extrabold text-primary">
            {item.title.slice(0, 1).toUpperCase()}
            {item.title.slice(1)}
          </h5>
        )}
        <span className="text-xl py-1 font-semibold text-primary">
          {item.type}
        </span>
        {item.description.length > 20 ? (
          <h5 className="text-base py-1 font-base text-primary">
            {item.description.slice(0, 1).toUpperCase()}
            {item.description.slice(1, 30)}...
          </h5>
        ) : (
          <h5 className="text-base py-1 font-base text-primary">
            {item.description.slice(0, 1).toUpperCase()}
            {item.description.slice(1)}
          </h5>
        )}
        <div className="">
          {typeof item.category === 'object' ? (
            item.category.slice(0, 3).map((cat) => {
              return (
                <span
                  key={cat}
                  className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg"
                >
                  {cat}
                </span>
              );
            })
          ) : (
            <span className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg">
              {item.category}
            </span>
          )}
          <h5 className="text-base py-1 font-base text-primary">
            {item.createdAt}
          </h5>
        </div>
        <button
          type="button"
          className="w-full my-2 duration-300 text-background bg-secondary hover:bg-primary focus:ring-1 focus:outline-none focus:ring-tertiary font-medium rounded-lg text-md px-5 py-2.5 text-center"
        >
          <a href={`mailto: ${item.owner?.email}`}>Email Owner</a>
        </button>
      </div>
    </div>
  );
}
export default ItemsCard;
