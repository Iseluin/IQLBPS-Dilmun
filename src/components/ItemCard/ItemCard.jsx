import { useState, useEffect } from 'react';
import { getItemList, itemList } from '../../features/slices/item';
import { useSelector, useDispatch } from '../../app/store';

function ItemCard() {
  const itemData = useSelector(itemList);
  const [items, setItems] = useState([]);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getItemList(itemData));
    setItems(itemData);
  }, []);
  return (
    <div data-testid='item-card' className="w-full grid gap-4 lg:grid-cols-3 md:grid-cols-2 grid-cols-1 content-center items-center justify-items-center bg-background py-6 my-2">
      {items.map((item) => {
        return (
          <div className="lg:w-80 md:w-64 w-96 bg-background m-7 rounded-lg hover:shadow-secondary shadow-gray-400 shadow-lg hover:shadow-2xl duration-300">
            <figure className="relative max-w-sm transition-all cursor-pointer">
              <a href="/">
                <img
                  className="rounded-t-lg overflow-hidden mb-3 lg:h-80 md:h-64 h-96 shadow-md md:w-64 lg:w-80 w-96"
                  src={item.file}
                  alt="product"
                />
                <figcaption className="absolute top-3 right-1 px-4 text-lg">
                  {item.type === 'Donated' ? (
                    <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                      {item.type}
                    </span>
                  ) : (
                    <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                      {item.price}
                    </span>
                  )}
                </figcaption>
              </a>
            </figure>
            <div className="px-5 pb-5">
              <h5 className="text-3xl pb-2 font-extrabold text-primary">
                {item.title.slice(0, 1).toUpperCase()}
                {item.title.slice(1)}
              </h5>
              <span className="text-xl py-1 font-semibold text-primary">
                {item.type}
              </span>
              {item.description.length > 20 ? (
                <h5 className="text-base py-1 font-base text-primary">
                  {item.description.slice(0, 1).toUpperCase()}
                  {item.description.slice(1, 30)}...
                </h5>
              ) : (
                <h5 className="text-base py-1 font-base text-primary">
                  {item.description.slice(0, 1).toUpperCase()}
                  {item.description.slice(1)}
                </h5>
              )}
              <div className="">
                {typeof item.category === 'object' ? (
                  item.category.map((cat) => {
                    return (
                      <span className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg">
                        {cat}
                      </span>
                    );
                  })
                ) : (
                  <span className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg">
                    {item.category}
                  </span>
                )}
                <h5 className="text-base py-1 font-base text-primary">
                  {item.createdAt.toString()}
                </h5>
              </div>
              <button
                type="button"
                className="w-full my-2 duration-300 text-background bg-secondary hover:bg-primary focus:ring-1 focus:outline-none focus:ring-tertiary font-medium rounded-lg text-md px-5 py-2.5 text-center"
              >
                Email Owner
              </button>
            </div>
          </div>
        );
      })}
    </div>
  );
}
export default ItemCard;
