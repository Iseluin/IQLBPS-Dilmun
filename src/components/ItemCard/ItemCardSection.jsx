import { useState, useEffect } from 'react';
import { useSelector, useDispatch } from '../../app/store';
import { getItemList, itemList } from '../../features/slices/item';
import ItemCard from './ItemCard';

export default function ItemCardSection() {
  const itemData = useSelector(itemList);
  const [items, setItems] = useState([]);
  const dispatch = useDispatch();

  const func = async () => {
    await dispatch(getItemList(itemData));
    setItems(itemData.docs.map((d) => {return ({...d.doc(), id: d.id} )}));
  };
  useEffect(() => {
    func();
  }, [items]);

  return (
    <div
      data-testid="item-card"
      className="w-full grid gap-4 lg:grid-cols-3 md:grid-cols-2 grid-cols-1 content-center items-center justify-items-center bg-background py-6 my-2"
    >
      <div>
        {items.map((item) => {
          return <ItemCard item={item} />;
        })}
      </div>
    </div>
  );
}
