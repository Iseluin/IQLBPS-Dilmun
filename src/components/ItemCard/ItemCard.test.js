import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import ItemCard from './ItemCard';

test('Sign In Page Renders Correctly', () => {
  render(
    <Provider store={store}>
      <ItemCard />
    </Provider>
  );
  expect(screen.getByTestId('item-card')).toMatchSnapshot();
});
