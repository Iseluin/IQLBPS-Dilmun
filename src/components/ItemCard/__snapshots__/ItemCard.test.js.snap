// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Page Renders Correctly 1`] = `
<div
  class="w-full grid gap-4 lg:grid-cols-3 md:grid-cols-2 grid-cols-1 content-center items-center justify-items-center bg-background py-6 my-2"
  data-testid="item-card"
>
  <div />
</div>
`;
